examples/playwright/src/theia-app-loader.ts

```diff
@@ -0,0 +1,167 @@
+    const workspace = initialWorkspace ? initialWorkspace : new TheiaWorkspace();
+    workspace.initialize();
+    return workspace;
+}
+
+export class TheiaBrowserAppLoader {
+
+    static async load<T extends TheiaApp>(
+        page: Page,
+        initialWorkspace?: TheiaWorkspace,
+        factory?: TheiaAppFactory<T>
+    ): Promise<T> {
+        const workspace = initializeWorkspace(initialWorkspace);
+        return this.createAndLoad<T>(page, workspace, factory);
+    }
+
+    protected static async createAndLoad<T extends TheiaApp>(
+        page: Page,
+        workspace: TheiaWorkspace,
+        factory?: TheiaAppFactory<T>
+    ): Promise<T> {
+        const appFactory = theiaAppFactory<T>(factory);
+        const app = new appFactory(page, workspace);
+        await this.loadOrReload(app, '/#' + app.workspace.urlEncodedPath);
+        await app.waitForShellAndInitialized();
+        return app;
+    }
+
+    protected static async loadOrReload(app: TheiaApp, url: string): Promise<void> {
+        if (app.page.url() === url) {
+            await app.page.reload();
+        } else {
+            const wasLoadedAlready = await app.isShellVisible();
+            await app.page.goto(url);
+            if (wasLoadedAlready) {
+                // Theia doesn't refresh on URL change only
+                // So we need to reload if the app was already loaded before
+                await app.page.reload();
+            }
+        }
+    }
+
+}
+
+export class TheiaElectronAppLoader {
+
+    static async load<T extends TheiaApp>(
+        launchOptions: ElectronLaunchOptions | object,
+        initialWorkspace?: TheiaWorkspace,
+        factory?: TheiaAppFactory<T>,
+    ): Promise<T> {
+        const workspace = initializeWorkspace(initialWorkspace);
+        const playwrightOptions = this.toPlaywrightOptions(launchOptions, initialWorkspace);
+        const electronApp = await electron.launch(playwrightOptions);
+        const page = await electronApp.firstWindow();
+        const appFactory = theiaAppFactory<T>(factory);
+        const app = new appFactory(page, workspace);
+        await app.waitForShellAndInitialized();
+        return app;
+    }
+
+    protected static toPlaywrightOptions(
+        electronLaunchOptions: ElectronLaunchOptions | object,
+        workspace?: TheiaWorkspace
+    ): object {
+        if (electronLaunchOptions instanceof ElectronLaunchOptions) {
+            return electronLaunchOptions.playwrightOptions(workspace);
+        }
+        return electronLaunchOptions;
+    }
+}
+
+export class ElectronLaunchOptions {
+
+    constructor(
+        protected readonly electronAppPath: string,
+        protected readonly pluginsPath?: string,
+        protected readonly additionalArgs: string[] = ['--no-cluster', '--no-native-window-frame']
+    ) { }
+
+    playwrightOptions(workspace?: TheiaWorkspace): object {
+        const executablePath = this.electronAppPath + '/node_modules/.bin/electron';
```

Comment:
Thank you! There were several issues with paths on Windows. Following your suggestion on using `path` instead of modifying strings, it should now work on Windows.

Reply:
Ecxept that now nothing works on Windows anymore, due to #12405 😬